subsys/bluetooth/host/shell/bt.c

```diff
@@ -595,15 +596,17 @@ static void scan_recv(const struct bt_le_scan_recv_info *info, struct net_buf_si
 
 			/* Use the above auto_connect.addr address to automatically connect */
 			if (auto_connect.connect_name) {
+				__maybe_unused int ercd;
+
 				auto_connect.connect_name = false;
 
-				cmd_scan_off(ctx_shell);
+				bt_do_scan_off();
 
 				/* "name" is what would be in argv[0] normally */
-				cmd_scan_filter_clear_name(ctx_shell, 1, (char *[]){"name"});
+				cmd_scan_filter_clear_name(NULL, 1, (char *[]){"name"});
```

Comment:
Ditto; this should be a function so we don't call command handlers with a `NULL` sh

Reply:
Got it